cmd/experimental/kjobctl/test/util/util.go

```diff
@@ -36,23 +45,98 @@ func DeleteNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace)
-		return err
-	}
-	if err := c.DeleteAllOf(ctx, &rayv1.RayJob{}, client.InNamespace(ns.Name)); err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	if err := c.DeleteAllOf(ctx, &rayv1.RayCluster{}, client.InNamespace(ns.Name)); err != nil && !apierrors.IsNotFound(err) {
+	if err := c.DeleteAllOf(ctx, &kjob.JobTemplate{}, client.InNamespace(ns.Name)); err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 	if err := c.Delete(ctx, ns); err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 	return nil
 }
+
+// Run executes the provided command within this context
+func Run(cmd *exec.Cmd) ([]byte, error) {
+	dir, _ := GetProjectDir()
+	cmd.Dir = dir
+
+	if err := os.Chdir(cmd.Dir); err != nil {
+		fmt.Fprintf(ginkgo.GinkgoWriter, "chdir dir: %s\n", err)
+	}
+
+	cmd.Env = append(os.Environ(), "GO111MODULE=on")
+	command := strings.Join(cmd.Args, " ")
+	fmt.Fprintf(ginkgo.GinkgoWriter, "running: %s\n", command)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return output, fmt.Errorf("%s failed with error: (%v) %s", command, err, string(output))
+	}
+
+	return output, nil
+}
+
+// GetNonEmptyLines converts given command output string into individual objects
+// according to line breakers, and ignores the empty elements in it.
+func GetNonEmptyLines(output string) []string {
+	var res []string
+	elements := strings.Split(output, "\n")
+	for _, element := range elements {
+		if element != "" {
+			res = append(res, element)
+		}
+	}
+
+	return res
+}
+
+// GetProjectDir will return the directory where the project is
+func GetProjectDir() (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return wd, err
+	}
+	wd = strings.ReplaceAll(wd, "/test/e2e", "")
+	return wd, nil
+}
+
+func CreateClientUsingCluster(kContext string) (client.Client, *rest.Config) {
+	cfg, err := config.GetConfigWithContext(kContext)
+	if err != nil {
+		fmt.Printf("unable to get kubeconfig for context %q: %s", kContext, err)
+		os.Exit(1)
+	}
+	gomega.Expect(cfg).NotTo(gomega.BeNil())
+
+	err = kjob.AddToScheme(scheme.Scheme)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+
+	err = corev1.AddToScheme(scheme.Scheme)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+
+	err = batchv1.AddToScheme(scheme.Scheme)
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+
+	client, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	return client, cfg
+}
+
+func IsKjobInstalled() {
+	cmd := exec.Command("kubectl", "get", "crds")
+	out, err := Run(cmd)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	crds := GetNonEmptyLines(string(out))
+	gomega.Expect(crds).NotTo(gomega.BeEmpty())
+
+	isKjobInstalled := slices.ContainsFunc(crds, func(s string) bool {
+		return strings.Contains(s, "kjobctl")
+	})
+	gomega.Expect(isKjobInstalled).Should(gomega.BeTrue())
+}
```

Comment:
Instead of having this check the default k8s context and do string manipulations to find the crd, if we 100% need this functionality , we should make it use a  provided cpntroller-runtime.Client (specific to the cluster we are running the tests into) and do structured `client.Get` or `client.List` to check if the CRD is created.

Reply:
I think it's not need. It should be already prepared by bash script.